exasol_sagemaker_extension/autopilot_prediction.py

```diff
@@ -15,19 +16,25 @@ def __init__(self, exa, model_connection_name,
     def run(self, ctx):
         model_connection = self.exa.get_connection(self.model_connection_name)
         endpoint_info_json = json.loads(model_connection.address)
-        aws_s3_conn_obj = self.exa.get_connection(
-            endpoint_info_json['aws_s3_connection'])
-
-        os.environ["AWS_DEFAULT_REGION"] = endpoint_info_json['aws_region']
-        os.environ["AWS_ACCESS_KEY_ID"] = aws_s3_conn_obj.user
-        os.environ["AWS_SECRET_ACCESS_KEY"] = aws_s3_conn_obj.password
 
         if endpoint_info_json['status'] == 'deployed':
-            data_df = ctx.get_dataframe()
+            aws_s3_conn_obj = self.exa.get_connection(
+                endpoint_info_json['aws_s3_connection'])
+
+            os.environ["AWS_DEFAULT_REGION"] = endpoint_info_json['aws_region']
+            os.environ["AWS_ACCESS_KEY_ID"] = aws_s3_conn_obj.user
+            os.environ["AWS_SECRET_ACCESS_KEY"] = aws_s3_conn_obj.password
+
+            data_df = ctx.get_dataframe(num_rows='all')
             pred = self.prediction_method(
-                endpoint_info_json['endpoint_name'].upper(), data_df)
+                endpoint_info_json['endpoint_name'], data_df)
 
-            data_df["predictions"] = pred
+            type = self.exa.meta.output_columns[-1].type
+            data_df["predictions"] = [type(ast.literal_eval(i)) for i in pred]
```

Comment:
Please not, eval in all its forms is dangerous, because the content of pred comes from an external source. What do you want to achieve here?